subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -591,7 +592,50 @@ public ImmutableAttributesFactory getAttributesCache() {
 
     @Override
     public void lockAttributes() {
-        configurationAttributes = configurationAttributes.asImmutable();
+        final AttributeContainerInternal delegatee = configurationAttributes.asImmutable();
+        configurationAttributes = new AttributeContainerInternal() {
```

Comment:
Might be worth extracting this into a private inner class to keep the method short.